pkg/mover/chart.go

```diff
@@ -236,17 +272,183 @@ func namespacedPath(fullpath, chartName string) string {
 }
 
 /*
-Move executes the Chart relocation which includes
+  Move perfomes the relocation.
+
+A regular move executes the Chart relocation which includes
 
 1 - Push all the images to their new locations
 
 2 - Rewrite the Helm Chart and its subcharts
 
 3 - Repackage the Helm chart as toChartFilename
+
+A move to tar will:
+
+1 - Drop all images locally
+
+2 - Pack also the original chart (unpacked) and the hints file
+
+3 - Package all in a single compressed tarball
 */
 func (cm *ChartMover) Move() error {
+	if cm.targetOfflineTar != "" {
+		return cm.archive()
+	}
+	return cm.moveChart()
+}
+
+func (cm *ChartMover) archive() error {
 	log := cm.logger
+	log.Printf("Archiving %s@%s...\n", cm.chart.Name(), cm.chart.Metadata.Version)
+
+	tarFolder := cm.targetOfflineTar + ".folder"
+	if err := os.MkdirAll(tarFolder, DefaultTarPermissions); err != nil {
+		return fmt.Errorf("failed to create tar folder %s: %w", tarFolder, err)
+	}
+	if err := archiveChart(tarFolder, cm.chart); err != nil {
+		return fmt.Errorf("failed tarring chart %s: %w", cm.chart.Name(), err)
+	}
+	if err := archiveImages(tarFolder, cm.imageChanges); err != nil {
+		return fmt.Errorf("failed tarring images: %w", err)
+	}
+	// TODO: hints file
+	return nil
+}
+
+func archiveChart(folder string, helmChart *chart.Chart) error {
+	chartSubdir := fmt.Sprintf("%s-%s.chart", helmChart.Name(), helmChart.Metadata.Version)
+	chartFolder := filepath.Join(folder, chartSubdir)
+	if err := os.MkdirAll(chartFolder, DefaultTarPermissions); err != nil {
+		return fmt.Errorf("failed to create chart subfolder %s: %w", chartFolder, err)
+	}
+	for _, f := range helmChart.Files {
+		filename := filepath.Join(chartFolder, f.Name)
+		dir := filepath.Dir(filename)
+		if dir != chartFolder {
+			if err := os.MkdirAll(dir, DefaultTarPermissions); err != nil {
+				return fmt.Errorf("failed to create %s: %w", dir, err)
+			}
+		}
+		if err := os.WriteFile(filename, f.Data, DefaultTarPermissions); err != nil {
+			return fmt.Errorf("failed to save %q: %w", filename, err)
+		}
+	}
+	return nil
+}
+
+func archiveImages(folder string, imageChanges []*internal.ImageChange) error {
+	containersFolder := filepath.Join(folder, "images")
+	for _, change := range imageChanges {
+		name := path.Base(change.ImageReference.Context().Name())
+		tag := change.ImageReference.Identifier()
+		imageName := fmt.Sprintf("%s-%s", name, tag)
+		imageFolder := filepath.Join(containersFolder, imageName)
+		if err := os.MkdirAll(imageFolder, DefaultTarPermissions); err != nil {
+			return fmt.Errorf("failed to create image subfolder %s: %w", imageFolder, err)
+		}
+		if err := archiveImage(imageFolder, change.Image); err != nil {
+			return fmt.Errorf("failed saving image %s: %w", imageName, err)
+		}
+		if err := archiveRepositoriesFile(imageFolder, change); err != nil {
+			return fmt.Errorf("Failed to save repositories file: %w", err)
+		}
+	}
+	return nil
+}
+
+func archiveImage(folder string, image v1.Image) error {
```

Comment:
I'd take a look at leveraging this library instead https://pkg.go.dev/github.com/google/go-containerregistry/pkg/v1/tarball#section-readme

Reply:
Done